### PR TITLE
Limit cipher-suites with TLS servers

### PR DESCRIPTION
**Pull request by scottgreenup**
We are required to limit our cipher suites to a limited subset as part of our internal guidelines. I've made this change in this PR.

### CODE REVIEW

File: pkg/util/tlsutil/config.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+			// tls.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384,            // ECDHE-RSA-AES256-SHA384
+			// tls.TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,    // ECDHE-ECDSA-CHACHA20-POLY1305
+			// tls.TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,    // ECDHE-RSA-CHACHA20-POLY1305
+
+			tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256, // ECDHE-ECDSA-AES128-SHA256
+			tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,    // ECDHE-ECDSA-AES128-SHA
+			tls.TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,    // ECDHE-ECDSA-AES256-SHA
+			tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256,   // ECDHE-RSA-AES128-SHA256
+			tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256, // ECDHE-ECDSA-AES128-GCM-SHA256
+			tls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384, // ECDHE-ECDSA-AES256-GCM-SHA384
```
Comment (brjacobsatlassian) [inline pkg/util/tlsutil/config.go]: OOC, why isn't this at the top?

File: pkg/util/tlsutil/config.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+		NextProtos: []string{"h2", "http/1.1"},
+
+		// TLS cipher suite preference must be dictated by the server
+		PreferServerCipherSuites: true,
+
+		// List of cipher suites comes from the standard.
+		// IANA names were mapped from testssl.sh and ciphersuite.info
+		CipherSuites: []uint16{
+			// These aren't supported in Go (yet), but are allowed by our standard.
+			// tls.TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384,          // ECDHE-ECDSA-AES256-SHA384
```
Comment (brjacobsatlassian) [inline pkg/util/tlsutil/config.go]: :cry: this is a surprising one to have Go not support

File: pkg/util/tlsutil/config.go (hunk 1 above)
Comment (brjacobsatlassian) [reply pkg/util/tlsutil/config.go]: ... wait a minute, this is a copy of line 26

Comment (scottgreenup) [reply pkg/util/tlsutil/config.go]: I totally forgot about ordering them :fearful:

**Comment by scottgreenup** [comment]
Do not merge, the list has changed.

File: pkg/util/tlsutil/config.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+		// IDs were mapped using `openssl ciphers -V` to get the code and then compared with the CSV available at
+		// https://www.iana.org/assignments/tls-parameters/tls-parameters-4.csv
+		//
+		// Those ciphers which are commented out are not supported in Go, they are here for completion to show exactly
+		// what is missing.
+		CipherSuites: []uint16{
+			tls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384, // ECDHE-ECDSA-AES256-GCM-SHA384
+			tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,   // ECDHE-RSA-AES256-GCM-SHA384
+
+			// tls.TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256, // ECDHE-ECDSA-CHACHA20-POLY1305
```
Comment (ychen-atlassian) [inline pkg/util/tlsutil/config.go]: Remove?

**Pull request merged**